src/dataframe/dataframe.jl

```diff
@@ -302,7 +302,10 @@ end
 
 # df[:, SingleColumnIndex] => AbstractVector
 # df[:, MultiColumnIndex] => DataFrame
-Base.getindex(df::DataFrame, row_ind::Colon, col_inds) = df[col_inds]
+function Base.getindex(df::DataFrame, row_ind::Colon, col_inds)
+    Base.depwarn("indexing with colon as row will create a copy of rows in the future", :getindex)
```

Comment:
"a copy" (not "of rows" more than "of columns"). Also point people at `df[col_inds]` to avoid the deprecation warning until we remove it.

Reply:
fixed